### ISSUE TITLE
Default cache_lifetime value for Fastly_CDN_Block_Cookie_Formkey block is incorrect

### ISSUE THREAD

**Issue by bneff84** [fastly/fastly-magento#50]
We identified a bug in this module where the cache_lifetime of the Fastly_CDN_Block_Cookie_Formkey block is set to `false` instead of `null` which causes the block to be cached indefinitely by Magento's block caching mechanism instead of never being cached with the `null` setting.

This causes a conflict on the very first (and only the first) page load of any session where Magento will grab the cached version of the block, which contains a formkey from another session. This causes the ESI replacement logic to fail to replace this formkey value with an <esi> tag when the final page is sent to the user by Fastly.

The below patch remedies this by changing the block to never cache, which allows the \Fastly_CDN_Model_Observer::replaceFormKeys logic to work correctly 100% of the time, even on the first page load of a session by insuring that the contents of the block will always contain the current session's formkey value.

```
--- app/code/community/Fastly/CDN/Block/Cookie/Formkey.php	(date 1585243890134)
+++ app/code/community/Fastly/CDN/Block/Cookie/Formkey.php	(date 1585243890134)
@@ -25,7 +25,7 @@
     {
         // set default cache lifetime and cache tags
         $this->addData(array(
-            'cache_lifetime'    => false
+            'cache_lifetime'    => null
        ));
     }
 ```

**Comment by smaeda-ks** [comment]
https://erfanimani.com/magento-block-caching-and-system-config

> A `cache_lifetime` of `false` will cause `cache_lifetime` to be set to the default value, which is 7200 or 2 hours.
> 
> A lifetime of `NULL` will cause the block to not be cached at all - this is useful for removing cache functionality from core or third party blocks. You can set cache lifetime to `NULL` through observer events or by rewriting blocks.

**Comment by bneff84** [comment]
@smaeda-ks that's correct and in this case you **should** be setting the cache to disabled for this block because of how the Fastly_CDN_Model_Observer::replaceFormKeys logic works. Since it searches the output HTML of the block for the current session's form_key value, if it uses block-level cache, it will not find that value, and thus not replace the form key with an ESI tag. This is the cause of the underlying issue we identified on several sites using Fastly on Magento with this module.

**Comment by smaeda-ks** [comment]
cc: @deninchoo @vvuksan

**Comment by vvuksan** [comment]
Fixed in https://github.com/fastly/fastly-magento/commit/ecaa2dcfd0a305a022b97762e8dd50318767b134 and released as version 1.0.31 of the plugin.

**Comment by smaeda-ks** [comment]
Ty!
